File: kal/src/main/java/com/jarvanmo/kal/recyclerview/paged/PagingNetworkState.java
package com.jarvanmo.kal.recyclerview.paged;

public class PagingNetworkState {
    public final static PagingNetworkState LOADED =new  PagingNetworkState(Status.SUCCESS);
    public final static PagingNetworkState LOADING = new PagingNetworkState(Status.RUNNING);
    public  static PagingNetworkState error(String msg) {
        return new PagingNetworkState(Status.FAILED, msg);
    }

    private Status status;
    private String msg;

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    private PagingNetworkState(Status status){
        this(status,null);
    }


    private PagingNetworkState(Status status,String msg){
        this.status = status;
        this.msg = msg;
    }
}

File: kal/src/main/java/com/jarvanmo/kal/recyclerview/paged/PagingNetworkState.kt
/*
 * Copyright (C) 2017 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.jarvanmo.kal.recyclerview.paged

enum class Status {
    RUNNING,
    SUCCESS,
    FAILED
}

@Suppress("DataClassPrivateConstructor")
data class PagingNetworkState private constructor(
        val status: Status,
        val msg: String? = null) {
    companion object {
        val LOADED = PagingNetworkState(Status.SUCCESS)
        val LOADING = PagingNetworkState(Status.RUNNING)
        fun error(msg: String?) = PagingNetworkState(Status.FAILED, msg)
    }
}